<?xml version="1.0" encoding="UTF-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro4sdf from model.sdf.xmacro           | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<sdf version="1.7">
	<!--include defination-->
	<!--model-->
	<model name="ur10_robotiq140">
		<!--model: ur10+robotiq140 -->
		<link name="base_link">
			<pose relative_to="__model__">0 0 0 0 -0 0</pose>
			<inertial>
				<mass>4</mass>
				<inertia>
					<ixx>0.00610633</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.00610633</iyy>
					<iyz>0</iyz>
					<izz>0.01125</izz>
				</inertia>
			</inertial>
			<collision name="base_link_collision">
				<geometry>
					<mesh>
						<uri>model://ur10/meshes/collision/base.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<visual name="base_link_visual">
				<geometry>
					<mesh>
						<uri>model://ur10/meshes/visual/base.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<self_collide>1</self_collide>
		</link>
		<joint name="shoulder_pan_joint" type="revolute">
			<pose relative_to="base_link">0 0 0.1273 0 -0 0</pose>
			<parent>base_link</parent>
			<child>shoulder_link</child>
			<axis>
				<xyz>0 0 1</xyz>
				<limit>
					<lower>-6.28319</lower>
					<upper>6.28319</upper>
					<effort>330</effort>
					<velocity>2.16</velocity>
				</limit>
				<dynamics>
					<damping>0</damping>
					<friction>0</friction>
					<spring_reference>0</spring_reference>
					<spring_stiffness>0</spring_stiffness>
				</dynamics>
			</axis>
		</joint>
		<link name="shoulder_link">
			<pose relative_to="shoulder_pan_joint">0 0 0 0 -0 0</pose>
			<inertial>
				<mass>7.778</mass>
				<inertia>
					<ixx>0.0314743</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.0314743</iyy>
					<iyz>0</iyz>
					<izz>0.0218756</izz>
				</inertia>
			</inertial>
			<collision name="shoulder_link_collision">
				<geometry>
					<mesh>
						<uri>model://ur10/meshes/collision/shoulder.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<visual name="shoulder_link_visual">
				<geometry>
					<mesh>
						<uri>model://ur10/meshes/visual/shoulder.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<self_collide>1</self_collide>
		</link>
		<joint name="shoulder_lift_joint" type="revolute">
			<pose relative_to="shoulder_link">0 0.220941 0 0 1.5708 0</pose>
			<parent>shoulder_link</parent>
			<child>upper_arm_link</child>
			<axis>
				<xyz>0 1 0</xyz>
				<limit>
					<lower>-6.28319</lower>
					<upper>6.28319</upper>
					<effort>330</effort>
					<velocity>2.16</velocity>
				</limit>
				<dynamics>
					<damping>0</damping>
					<friction>0</friction>
					<spring_reference>0</spring_reference>
					<spring_stiffness>0</spring_stiffness>
				</dynamics>
			</axis>
		</joint>
		<link name="upper_arm_link">
			<pose relative_to="shoulder_lift_joint">0 0 0 0 -0 0</pose>
			<inertial>
				<pose>0 -0.045 0.306 0 -0 0</pose>
				<mass>12.93</mass>
				<inertia>
					<ixx>0.421754</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.421754</iyy>
					<iyz>0</iyz>
					<izz>0.0363656</izz>
				</inertia>
			</inertial>
			<collision name="upper_arm_link_collision">
				<geometry>
					<mesh>
						<uri>model://ur10/meshes/collision/upperarm.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<visual name="upper_arm_link_visual">
				<geometry>
					<mesh>
						<uri>model://ur10/meshes/visual/upperarm.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<self_collide>1</self_collide>
		</link>
		<joint name="elbow_joint" type="revolute">
			<pose relative_to="upper_arm_link">0 -0.1719 0.612 0 -0 0</pose>
			<parent>upper_arm_link</parent>
			<child>forearm_link</child>
			<axis>
				<xyz>0 1 0</xyz>
				<limit>
					<lower>-3.14159</lower>
					<upper>3.14159</upper>
					<effort>150</effort>
					<velocity>3.15</velocity>
				</limit>
				<dynamics>
					<damping>0</damping>
					<friction>0</friction>
					<spring_reference>0</spring_reference>
					<spring_stiffness>0</spring_stiffness>
				</dynamics>
			</axis>
		</joint>
		<link name="forearm_link">
			<pose relative_to="elbow_joint">0 0 0 0 -0 0</pose>
			<inertial>
				<pose>0 0 0.28615 0 -0 0</pose>
				<mass>3.87</mass>
				<inertia>
					<ixx>0.11107</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.11107</iyy>
					<iyz>0</iyz>
					<izz>0.0108844</izz>
				</inertia>
			</inertial>
			<collision name="forearm_link_collision">
				<geometry>
					<mesh>
						<uri>model://ur10/meshes/collision/forearm.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<visual name="forearm_link_visual">
				<geometry>
					<mesh>
						<uri>model://ur10/meshes/visual/forearm.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<self_collide>1</self_collide>
		</link>
		<joint name="wrist_1_joint" type="revolute">
			<pose relative_to="forearm_link">0 0 0.5723 0 1.5708 0</pose>
			<parent>forearm_link</parent>
			<child>wrist_1_link</child>
			<axis>
				<xyz>0 1 0</xyz>
				<limit>
					<lower>-6.28319</lower>
					<upper>6.28319</upper>
					<effort>54</effort>
					<velocity>3.2</velocity>
				</limit>
				<dynamics>
					<damping>0</damping>
					<friction>0</friction>
					<spring_reference>0</spring_reference>
					<spring_stiffness>0</spring_stiffness>
				</dynamics>
			</axis>
		</joint>
		<link name="wrist_1_link">
			<pose relative_to="wrist_1_joint">0 0 0 0 -0 0</pose>
			<inertial>
				<pose>0 0.1149 0 0 -0 0</pose>
				<mass>1.96</mass>
				<inertia>
					<ixx>0.00510825</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.00510825</iyy>
					<iyz>0</iyz>
					<izz>0.0055125</izz>
				</inertia>
			</inertial>
			<collision name="wrist_1_link_collision">
				<geometry>
					<mesh>
						<uri>model://ur10/meshes/collision/wrist1.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<visual name="wrist_1_link_visual">
				<geometry>
					<mesh>
						<uri>model://ur10/meshes/visual/wrist1.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<self_collide>1</self_collide>
		</link>
		<joint name="wrist_2_joint" type="revolute">
			<pose relative_to="wrist_1_link">0 0.1149 0 0 -0 0</pose>
			<parent>wrist_1_link</parent>
			<child>wrist_2_link</child>
			<axis>
				<xyz>0 0 1</xyz>
				<limit>
					<lower>-6.28319</lower>
					<upper>6.28319</upper>
					<effort>54</effort>
					<velocity>3.2</velocity>
				</limit>
				<dynamics>
					<damping>0</damping>
					<friction>0</friction>
					<spring_reference>0</spring_reference>
					<spring_stiffness>0</spring_stiffness>
				</dynamics>
			</axis>
		</joint>
		<link name="wrist_2_link">
			<pose relative_to="wrist_2_joint">0 0 0 0 -0 0</pose>
			<inertial>
				<pose>0 0 0.1157 0 -0 0</pose>
				<mass>1.96</mass>
				<inertia>
					<ixx>0.00510825</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.00510825</iyy>
					<iyz>0</iyz>
					<izz>0.0055125</izz>
				</inertia>
			</inertial>
			<collision name="wrist_2_link_collision">
				<geometry>
					<mesh>
						<uri>model://ur10/meshes/collision/wrist2.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<visual name="wrist_2_link_visual">
				<geometry>
					<mesh>
						<uri>model://ur10/meshes/visual/wrist2.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<self_collide>1</self_collide>
		</link>
		<joint name="wrist_3_joint" type="revolute">
			<pose relative_to="wrist_2_link">0 0 0.1157 0 -0 0</pose>
			<parent>wrist_2_link</parent>
			<child>wrist_3_link</child>
			<axis>
				<xyz>0 1 0</xyz>
				<limit>
					<lower>-6.28319</lower>
					<upper>6.28319</upper>
					<effort>54</effort>
					<velocity>3.2</velocity>
				</limit>
				<dynamics>
					<damping>0</damping>
					<friction>0</friction>
					<spring_reference>0</spring_reference>
					<spring_stiffness>0</spring_stiffness>
				</dynamics>
			</axis>
		</joint>
		<link name="wrist_3_link">
			<pose relative_to="wrist_3_joint">0 0 0 0 -0 0</pose>
			<inertial>
				<pose>0 0.07695 0 1.5708 -0 0</pose>
				<mass>0.202</mass>
				<inertia>
					<ixx>0.000117922</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.000117922</iyy>
					<iyz>0</iyz>
					<izz>0.000204525</izz>
				</inertia>
			</inertial>
			<collision name="wrist_3_link_collision">
				<geometry>
					<mesh>
						<uri>model://ur10/meshes/collision/wrist3.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<visual name="wrist_3_link_visual">
				<geometry>
					<mesh>
						<uri>model://ur10/meshes/visual/wrist3.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<self_collide>1</self_collide>
		</link>
		<link name="robotiq_arg2f_base_link">
			<pose relative_to="ur10_robotiq140_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0 -5e-06 0.03145 0 -0 0</pose>
				<mass>0.22652</mass>
				<inertia>
					<ixx>0.00020005</ixx>
					<ixy>-4.2442e-10</ixy>
					<ixz>-2.9069e-10</ixz>
					<iyy>0.00017832</iyy>
					<iyz>-3.4402e-08</iyz>
					<izz>0.00013478</izz>
				</inertia>
			</inertial>
			<collision name="robotiq_arg2f_base_link_collision">
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/collision/robotiq_arg2f_base_link.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<visual name="robotiq_arg2f_base_link_visual">
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/visual/robotiq_arg2f_base_link.stl</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0.1 0.1 0.1 1</ambient>
					<diffuse>0.1 0.1 0.1 1</diffuse>
					<specular>1 1 1 1</specular>
				</material>
			</visual>
		</link>
		<joint name="finger_joint" type="revolute">
			<pose relative_to="robotiq_arg2f_base_link">0 -0.030601 0.054905 2.2958 -0 0</pose>
			<parent>robotiq_arg2f_base_link</parent>
			<child>left_outer_knuckle</child>
			<axis>
				<xyz>-1 0 0</xyz>
				<limit>
					<lower>0</lower>
					<upper>0.7</upper>
					<effort>1000</effort>
					<velocity>2</velocity>
				</limit>
				<dynamics>
					<spring_reference>0</spring_reference>
					<spring_stiffness>0</spring_stiffness>
				</dynamics>
			</axis>
		</joint>
		<link name="left_outer_knuckle">
			<pose relative_to="finger_joint">0 0 0 0 -0 0</pose>
			<inertial>
				<pose>0.000164 0.04584 0.01178 0 -0 0</pose>
				<mass>0.0311462</mass>
				<inertia>
					<ixx>2.96016e-05</ixx>
					<ixy>-1.09755e-07</ixy>
					<ixz>7.40449e-08</ixz>
					<iyy>1.16215e-05</iyy>
					<iyz>1.08762e-05</iyz>
					<izz>2.07508e-05</izz>
				</inertia>
			</inertial>
			<collision name="left_outer_knuckle_collision">
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/collision/robotiq_arg2f_140_outer_knuckle.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<collision name="left_outer_finger_collision">
				<pose>0 0.01822 0.026002 0 -0 0</pose>
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/collision/robotiq_arg2f_140_outer_finger.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<visual name="left_outer_knuckle_visual">
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/visual/robotiq_arg2f_140_outer_knuckle.stl</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0.7 0.7 0.7 1</ambient>
					<diffuse>0.7 0.7 0.7 1</diffuse>
					<specular>1 1 1 1</specular>
				</material>
			</visual>
			<visual name="left_outer_finger_visual">
				<pose>0 0.01822 0.026002 0 -0 0</pose>
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/visual/robotiq_arg2f_140_outer_finger.stl</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0.3 0.3 0.3 1</ambient>
					<diffuse>0.3 0.3 0.3 1</diffuse>
					<specular>1 1 1 1</specular>
				</material>
			</visual>
		</link>
		<joint name="left_inner_finger_joint" type="revolute">
			<pose relative_to="left_outer_knuckle">0 0.099975 -0.002219 -0.725 0 0</pose>
			<parent>left_outer_knuckle</parent>
			<child>left_inner_finger</child>
			<axis>
				<xyz>1 0 0</xyz>
				<limit>
					<lower>0</lower>
					<upper>0.7</upper>
					<effort>1000</effort>
					<velocity>2</velocity>
				</limit>
				<dynamics>
					<spring_reference>0</spring_reference>
					<spring_stiffness>0</spring_stiffness>
				</dynamics>
			</axis>
		</joint>
		<link name="left_inner_finger">
			<pose relative_to="left_inner_finger_joint">0 0 0 0 -0 0</pose>
			<inertial>
				<pose>0.0003 0.016008 -0.013695 0 -0 0</pose>
				<mass>0.0104003</mass>
				<inertia>
					<ixx>2.71909e-06</ixx>
					<ixy>1.35402e-21</ixy>
					<ixz>-7.18173e-22</ixz>
					<iyy>7.691e-07</iyy>
					<iyz>6.74715e-07</iyz>
					<izz>2.30315e-06</izz>
				</inertia>
			</inertial>
			<collision name="left_inner_finger_collision">
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/collision/robotiq_arg2f_140_inner_finger.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<collision name="left_inner_finger_pad_collision">
				<pose>0 0.045755 -0.02722 0 -0 0</pose>
				<geometry>
					<box>
						<size>0.03 0.07 0.0075</size>
					</box>
				</geometry>
				<surface>
					<friction>
						<ode>
							<mu>1.05</mu>
							<mu2>1.05</mu2>
						</ode>
					</friction>
				</surface>
			</collision>
			<visual name="left_inner_finger_visual">
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/visual/robotiq_arg2f_140_inner_finger.stl</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0.3 0.3 0.3 1</ambient>
					<diffuse>0.3 0.3 0.3 1</diffuse>
					<specular>1 1 1 1</specular>
				</material>
			</visual>
			<visual name="left_inner_finger_pad_visual">
				<pose>0 0.045755 -0.02722 0 -0 0</pose>
				<geometry>
					<box>
						<size>0.027 0.065 0.0075</size>
					</box>
				</geometry>
				<material>
					<ambient>0.7 0.7 0.7 1</ambient>
					<diffuse>0.7 0.7 0.7 1</diffuse>
					<specular>1 1 1 1</specular>
				</material>
			</visual>
		</link>
		<joint name="left_inner_knuckle_joint" type="revolute">
			<pose relative_to="robotiq_arg2f_base_link">0 -0.0127 0.06142 2.2958 -0 0</pose>
			<parent>robotiq_arg2f_base_link</parent>
			<child>left_inner_knuckle</child>
			<axis>
				<xyz>1 0 0</xyz>
				<limit>
					<lower>-0.7</lower>
					<upper>0</upper>
					<effort>1000</effort>
					<velocity>2</velocity>
				</limit>
				<dynamics>
					<spring_reference>0</spring_reference>
					<spring_stiffness>0</spring_stiffness>
				</dynamics>
			</axis>
		</joint>
		<link name="left_inner_knuckle">
			<pose relative_to="left_inner_knuckle_joint">0 0 0 0 -0 0</pose>
			<inertial>
				<pose>0.000123 0.050785 0.00104 0 -0 0</pose>
				<mass>0.0271177</mass>
				<inertia>
					<ixx>2.6191e-05</ixx>
					<ixy>-2.43617e-07</ixy>
					<ixz>-6.3779e-09</ixz>
					<iyy>2.82702e-06</iyy>
					<iyz>-5.37201e-07</iyz>
					<izz>2.83696e-05</izz>
				</inertia>
			</inertial>
			<collision name="left_inner_knuckle_collision">
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/collision/robotiq_arg2f_140_inner_knuckle.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<visual name="left_inner_knuckle_visual">
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/visual/robotiq_arg2f_140_inner_knuckle.stl</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0.3 0.3 0.3 1</ambient>
					<diffuse>0.3 0.3 0.3 1</diffuse>
					<specular>1 1 1 1</specular>
				</material>
			</visual>
		</link>
		<joint name="right_inner_knuckle_joint" type="revolute">
			<pose relative_to="robotiq_arg2f_base_link">0 0.0127 0.06142 2.2958 0 -3.14159</pose>
			<parent>robotiq_arg2f_base_link</parent>
			<child>right_inner_knuckle</child>
			<axis>
				<xyz>1 0 0</xyz>
				<limit>
					<lower>-0.7</lower>
					<upper>0</upper>
					<effort>1000</effort>
					<velocity>2</velocity>
				</limit>
				<dynamics>
					<spring_reference>0</spring_reference>
					<spring_stiffness>0</spring_stiffness>
				</dynamics>
			</axis>
		</joint>
		<link name="right_inner_knuckle">
			<pose relative_to="right_inner_knuckle_joint">0 0 0 0 -0 0</pose>
			<inertial>
				<pose>0.000123 0.050785 0.00104 0 -0 0</pose>
				<mass>0.0271177</mass>
				<inertia>
					<ixx>2.6191e-05</ixx>
					<ixy>-2.43617e-07</ixy>
					<ixz>-6.3779e-09</ixz>
					<iyy>2.82702e-06</iyy>
					<iyz>-5.37201e-07</iyz>
					<izz>2.83696e-05</izz>
				</inertia>
			</inertial>
			<collision name="right_inner_knuckle_collision">
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/collision/robotiq_arg2f_140_inner_knuckle.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<visual name="right_inner_knuckle_visual">
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/visual/robotiq_arg2f_140_inner_knuckle.stl</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0.3 0.3 0.3 1</ambient>
					<diffuse>0.3 0.3 0.3 1</diffuse>
					<specular>1 1 1 1</specular>
				</material>
			</visual>
		</link>
		<joint name="right_outer_knuckle_joint" type="revolute">
			<pose relative_to="robotiq_arg2f_base_link">0 0.030601 0.054905 2.2958 -0 3.14159</pose>
			<parent>robotiq_arg2f_base_link</parent>
			<child>right_outer_knuckle</child>
			<axis>
				<xyz>1 0 0</xyz>
				<limit>
					<lower>-0.7</lower>
					<upper>0</upper>
					<effort>1000</effort>
					<velocity>2</velocity>
				</limit>
				<dynamics>
					<spring_reference>0</spring_reference>
					<spring_stiffness>0</spring_stiffness>
				</dynamics>
			</axis>
		</joint>
		<link name="right_outer_knuckle">
			<pose relative_to="right_outer_knuckle_joint">0 0 0 0 -0 0</pose>
			<inertial>
				<pose>0.000164 0.04584 0.01178 0 -0 0</pose>
				<mass>0.0311462</mass>
				<inertia>
					<ixx>2.96016e-05</ixx>
					<ixy>-1.09755e-07</ixy>
					<ixz>7.40449e-08</ixz>
					<iyy>1.16215e-05</iyy>
					<iyz>1.08762e-05</iyz>
					<izz>2.07508e-05</izz>
				</inertia>
			</inertial>
			<collision name="right_outer_knuckle_collision">
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/collision/robotiq_arg2f_140_outer_knuckle.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<collision name="right_outer_finger_collision">
				<pose>0 0.01822 0.026002 0 -0 0</pose>
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/collision/robotiq_arg2f_140_outer_finger.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<visual name="right_outer_knuckle_visual">
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/visual/robotiq_arg2f_140_outer_knuckle.stl</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0.7 0.7 0.7 1</ambient>
					<diffuse>0.7 0.7 0.7 1</diffuse>
					<specular>1 1 1 1</specular>
				</material>
			</visual>
			<visual name="right_outer_finger_visual">
				<pose>0 0.01822 0.026002 0 -0 0</pose>
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/visual/robotiq_arg2f_140_outer_finger.stl</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0.3 0.3 0.3 1</ambient>
					<diffuse>0.3 0.3 0.3 1</diffuse>
					<specular>1 1 1 1</specular>
				</material>
			</visual>
		</link>
		<joint name="right_inner_finger_joint" type="revolute">
			<pose relative_to="right_outer_knuckle">0 0.099975 -0.002219 -0.725 0 0</pose>
			<parent>right_outer_knuckle</parent>
			<child>right_inner_finger</child>
			<axis>
				<xyz>1 0 0</xyz>
				<limit>
					<lower>0</lower>
					<upper>0.7</upper>
					<effort>1000</effort>
					<velocity>2</velocity>
				</limit>
				<dynamics>
					<spring_reference>0</spring_reference>
					<spring_stiffness>0</spring_stiffness>
				</dynamics>
			</axis>
		</joint>
		<link name="right_inner_finger">
			<pose relative_to="right_inner_finger_joint">0 0 0 0 -0 0</pose>
			<inertial>
				<pose>0.0003 0.016008 -0.013695 0 -0 0</pose>
				<mass>0.0104003</mass>
				<inertia>
					<ixx>2.71909e-06</ixx>
					<ixy>1.35402e-21</ixy>
					<ixz>-7.18173e-22</ixz>
					<iyy>7.691e-07</iyy>
					<iyz>6.74715e-07</iyz>
					<izz>2.30315e-06</izz>
				</inertia>
			</inertial>
			<collision name="right_inner_finger_collision">
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/collision/robotiq_arg2f_140_inner_finger.stl</uri>
					</mesh>
				</geometry>
			</collision>
			<collision name="right_inner_finger_pad_collision">
				<pose>0 0.045755 -0.02722 0 -0 0</pose>
				<geometry>
					<box>
						<size>0.03 0.07 0.0075</size>
					</box>
				</geometry>
				<surface>
					<friction>
						<ode>
							<mu>1.05</mu>
							<mu2>1.05</mu2>
						</ode>
					</friction>
				</surface>
			</collision>
			<visual name="right_inner_finger_visual">
				<geometry>
					<mesh>
						<uri>model://robotiq140/meshes/visual/robotiq_arg2f_140_inner_finger.stl</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0.3 0.3 0.3 1</ambient>
					<diffuse>0.3 0.3 0.3 1</diffuse>
					<specular>1 1 1 1</specular>
				</material>
			</visual>
			<visual name="right_inner_finger_pad_visual">
				<pose>0 0.045755 -0.02722 0 -0 0</pose>
				<geometry>
					<box>
						<size>0.027 0.065 0.0075</size>
					</box>
				</geometry>
				<material>
					<ambient>0.7 0.7 0.7 1</ambient>
					<diffuse>0.7 0.7 0.7 1</diffuse>
					<specular>1 1 1 1</specular>
				</material>
			</visual>
		</link>
		<joint name="ur10_robotiq140_joint" type="fixed">
			<pose relative_to="wrist_3_link">0 0.0922 0 -1.570795 -1.570795 0</pose>
			<parent>wrist_3_link</parent>
			<child>robotiq_arg2f_base_link</child>
		</joint>
		<joint name="world_ur10_joint" type="fixed">
			<parent>world</parent>
			<child>base_link</child>
		</joint>
		<!--controller plugins for ur10-->
		<plugin filename="ignition-gazebo-joint-position-controller-system" name="ignition::gazebo::systems::JointPositionController">
			<joint_name>shoulder_pan_joint</joint_name>
			<p_gain>10000</p_gain>
			<i_gain>100</i_gain>
			<d_gain>500</d_gain>
			<i_max>200.0</i_max>
			<i_min>-200.0</i_min>
		</plugin>
		<plugin filename="ignition-gazebo-joint-position-controller-system" name="ignition::gazebo::systems::JointPositionController">
			<joint_name>shoulder_lift_joint</joint_name>
			<p_gain>50000</p_gain>
			<i_gain>100</i_gain>
			<d_gain>1000</d_gain>
			<i_max>200.0</i_max>
			<i_min>-200.0</i_min>
		</plugin>
		<plugin filename="ignition-gazebo-joint-position-controller-system" name="ignition::gazebo::systems::JointPositionController">
			<joint_name>elbow_joint</joint_name>
			<p_gain>50000.0</p_gain>
			<i_gain>100</i_gain>
			<d_gain>1000</d_gain>
			<i_max>200</i_max>
			<i_min>-200</i_min>
		</plugin>
		<plugin filename="ignition-gazebo-joint-position-controller-system" name="ignition::gazebo::systems::JointPositionController">
			<joint_name>wrist_1_joint</joint_name>
			<p_gain>100.0</p_gain>
			<i_gain>50.0</i_gain>
			<d_gain>10.0</d_gain>
			<i_max>10.0</i_max>
			<i_min>-10.0</i_min>
		</plugin>
		<plugin filename="ignition-gazebo-joint-position-controller-system" name="ignition::gazebo::systems::JointPositionController">
			<joint_name>wrist_2_joint</joint_name>
			<p_gain>75.0</p_gain>
			<i_gain>10</i_gain>
			<d_gain>5</d_gain>
			<i_max>10.0</i_max>
			<i_min>-10.0</i_min>
		</plugin>
		<plugin filename="ignition-gazebo-joint-position-controller-system" name="ignition::gazebo::systems::JointPositionController">
			<joint_name>wrist_3_joint</joint_name>
			<p_gain>25.0</p_gain>
			<i_gain>10</i_gain>
			<d_gain>0.1</d_gain>
			<i_max>10.0</i_max>
			<i_min>-10.0</i_min>
		</plugin>
		<!--controller plugins for robotiq140-->
		<plugin filename="RobotiqController" name="ignition::gazebo::systems::RobotiqController">
        </plugin>
		<!--joint state publisher-->
		<plugin filename="ignition-gazebo-joint-state-publisher-system" name="ignition::gazebo::systems::JointStatePublisher">
			<joint_name>shoulder_pan_joint</joint_name>
			<joint_name>shoulder_lift_joint</joint_name>
			<joint_name>elbow_joint</joint_name>
			<joint_name>wrist_1_joint</joint_name>
			<joint_name>wrist_2_joint</joint_name>
			<joint_name>wrist_3_joint</joint_name>
			<joint_name>finger_joint</joint_name>
			<joint_name>left_inner_finger_joint</joint_name>
			<joint_name>left_inner_knuckle_joint</joint_name>
			<joint_name>right_inner_knuckle_joint</joint_name>
			<joint_name>right_outer_knuckle_joint</joint_name>
			<joint_name>right_inner_finger_joint</joint_name>
		</plugin>
	</model>
</sdf>
